Deprecate feature flag for Log threshold alert details page

benakansara opened the pull request:
Resolves https://github.com/elastic/kibana/issues/172379

- Deprecates following feature flag used for enabling/disabling Log threshold alert details page: 
```
xpack.observability.unsafe.alertDetails.logs.enabled
```
- Removes usage of this flag from code.
- Adding this flag in `kibana.yml` will generate following warning:
```
[WARN ][config.deprecation] You no longer need to configure "xpack.observability.unsafe.alertDetails.logs.enabled".
```

<!--ONMERGE {"backportTargets":["8.12"]} ONMERGE-->

kibanamachine committed:
skip failing test suite (#165079)

kibanamachine committed:
skip failing test suite (#167496)

benakansara committed:
deprecate feature flag

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

benakansara committed:
remove feature flag from test suites

benakansara committed:
remove feature flag usage from tests

benakansara committed:
add config back as it is present in schema

jbudz commented on src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker:
```diff
@@ -306,7 +306,6 @@ kibana_vars=(
     xpack.ingestManager.registryUrl
     xpack.observability.annotations.index
     xpack.observability.unsafe.alertDetails.metrics.enabled
-    xpack.observability.unsafe.alertDetails.logs.enabled
```
It's used at https://github.com/elastic/kibana/pull/172554/files#diff-53c98e8d9e2d61d5c9f37b003fc6de5c3f84570f7886cf4bdd89f596b32a15c9R73.

In a docker environment, without this instead of getting a deprecation warning the configuration will be ignored.  It's a wrapper around what environment variables are passed to bin/kibana.

jbudz replied:
Are we okay with skipping the deprecation warning?

benakansara replied:
The reason I didn't remove the config is to prevent kibana crashing if user has discovered and used config (following recommendation from [this document](https://docs.google.com/document/d/1J0MggFsHYXE3qkPnqw-GgrznbMuWizRdQdHHqyd3oVA/edit)).

The deprecation warning is nice to have but not a must. As long as it doesn't break kibana and silently ignores the config in docker environment, it should be fine.

benakansara replied:
@jbudz I've added the feature flag back in `docker-kibana` ff0991caae5a07550c07f5f910d94e02f5e9a965. I will remove it when the config is removed as per your suggestion.

maryam-saeidi replied:
@benakansara Was there another separate discussion about this topic? I am trying to understand why we need to pass this config instead of ignoring it and showing a warning.

azasypkin commented:
(removed review request for @elastic/kibana-security as it's no longer required after https://github.com/elastic/kibana/pull/172554/commits/114fcf44ebd579183d4d8f6e658dff4a9a9f9e05)

maryam-saeidi commented on x-pack/plugins/observability/public/pages/overview/overview.stories.tsx:
```diff
@@ -78,17 +78,16 @@ const withCore = makeDecorator({
     } as unknown as Partial<CoreStart>);
 
-    const config: ConfigSchema = {
+    const config = {
```
What is this change about? Casting objects with `as` is just forcing typescript to ignore this type checking.

benakansara replied:
Made `logs` optional in config schema.

maryam-saeidi reviewed (commented):
Tested locally and got the warning when I had the FF and enabling/disabling it didn't have an impact on showing the alert details page 👍🏻 

Just added one question about type casting.

<img src="https://github.com/elastic/kibana/assets/12370520/eca80b7c-ae2f-4f35-9365-590ee645686f" width=500 />

benakansara committed:
adding config back in docker-kibana

benakansara committed:
make logs config optional

benakansara committed:
fix types

maryam-saeidi reviewed (commented):
Can we also remove this one? https://github.com/elastic/kibana/blob/main/test/plugin_functional/test_suites/core_plugins/rendering.ts#L328

benakansara commented:
> Can we also remove this one? https://github.com/elastic/kibana/blob/main/test/plugin_functional/test_suites/core_plugins/rendering.ts#L328

I had removed it before and got this failure: https://buildkite.com/elastic/kibana-pull-request/builds/181747#018c3a00-ef61-4dc0-84a2-53b96c9671c7
```
Error: Actual config settings exposed to the browser do not match list defined in test; this assertion fails if extra settings are present and/or expected settings are missing
```

I added it back because we are not removing the feature flag from config, but deprecating it. This should be removed when the flag is removed from the config in future.

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.12|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/172775">](https://github.com/elastic/kibana/pull/172775)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.7 -->
